src/foam/nanos/approval/ApprovableAwareDAO.js

```diff
@@ -375,112 +324,52 @@ foam.CLASS({
         }
 
         // then handle the diff here and attach it into the approval request
-        Map updatedProperties = currentObjectInDAO.diff(obj);
+        FObject objectToDiffAgainst = currentObjectInDAO;
+
+        // diffing against an empty object gives us the properties in map form for create
+        if ( operation == Operations.CREATE ){
+          try {
+            objectToDiffAgainst = (FObject) obj.getClassInfo().newInstance();
+          } catch (Exception e){
+            throw new RuntimeException(e);
+          }
+        }
+
+        Map propertiesToApprove = objectToDiffAgainst.diff(obj);
+        
 
         for ( int i = 0; i < storageTransientPropertyNames.size(); i++ ){
-          updatedProperties.remove(storageTransientPropertyNames.get(i));
+          propertiesToApprove.remove(storageTransientPropertyNames.get(i));
```

Comment:
@jlhughes @nanoscott as mentioned above I can remove this part which takes the storageTransient properties out of the propertiesToUpdate map. But keep the removal of network and storage transient when generating the lookup key and hide storage transient and network transient properties in the PropertiesToUpdateView.

Reply:
Yes, remove for lookup but keep for update.  got it.